### DISCUSSION TITLE
Zeitschaltung für den Dynamic Power Limiter

### DISCUSSION THREAD

**Discussion by Tomv888** [hoylabs/OpenDTU-OnBattery#936] [Ideas]
Eine Frage an die Programmierer,
Wäre es möglich,dem Dynamic Power Limiter eine Zeitregel hinzuzufügen?
zB so,das er nur von 10-18 uhr aktiv ist?

**Comment by genmad** [comment]
Was willst du dann in der restlichen Zeit machen?

**Comment by Tomv888** [comment]
in der restlichen Zeit soll der Umrichter ohne Leistungsbegrenzung,also auf 100% laufen

**Comment by greymda** [comment]
i think you can use HA with MQQT to enable/disable DPL based on your criteria

**Comment by Tomv888** [comment]
leider nutze ich kein HA

**Comment by Manos1966** [comment]
@Tomv888 vielleicht solltest du versuchen zu beschreiben was genau du erreichen willst....

> in der restlichen Zeit soll der Umrichter ohne Leistungsbegrenzung, **also auf 100% laufen**

Der DPL wurde entwickelt mit Optionen, die genau das ermoeglichen: 
- **Solar Pass-through:** Tagsueber regelt mit Prioritaet, die Nulleinspeisung zu erreichen (und falls etwas uebrig bleibt, wird fuer das Laden der Batterie benutzt)
- **Full Solar Pass-through:** Erreicht die Batterie eine gewisse Kapazitaet % (hier 90%), wechselt der DPL auf Full Solar Pass-through: **Die gesamte Solar Produktion wird ins Netz eingespeist.** 
![Full Solar PassThrough](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/0c9d821c-3ad3-4303-b117-124e97146c84)

**Comment by Tomv888** [comment]
ich nutze den HMS2000-4T. 3 Module sind nach Süden ausgerichtet,das 4. ist auf einem Tracker,der von Ost bis West fährt.
Wenn der Tracker nun auf West steht,sind die Südmodule schon lange verschattet.gemäß meiner Einstellungen im DPL
regelt die DTU den HMS nun zb auf 800W.Das bedeutet,das jeder der 4 MPPT auf 200W begrenzt ist,also auch das Westmodul auf dem Tracker.
Dieses könnte aber ohne Begrenzung viel mehr Leistung generieren und so einen Energiebezug aus dem Netz auf "später" verschieben.

Darum die Frage nach einem Timer für den DPL.

**Reply by spcqike** [reply]
Ah. Ein typisches XY-Problem :) 

Du suchst eigentlich keine zeitliche Regelung, sondern sowas

https://github.com/helgeerbe/OpenDTU-OnBattery/issues/917

gibts aber leider auch nicht

**Comment by Manos1966** [comment]
@Tomv888 Ich sehe das Problem.

Wie spcqike erklaert hat...
Deswegen heisst dieses Projekt OpenDTU-**onBattery** 
Die ganze Entwicklung ist hauptsächlich mit Blick auf die viel komplexere Steuerung einer Batterie entwickelt.

Die Kombination von "Hoymiles-Einschrenkungen" (= die MPPTs) und der Tatsache, dass eine an der Batterie angeschlossene Hoymiles gleiche Stroeme auf alle Eingaenge erhaelt, machen die Situation so wie sie ist.

Eine Loesung gibt es vielleicht doch, aber "Ohne Fleisch, kein Preis"
Werfe ein Auge hier: https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/Web-API#example-2-change-power-limit
(und danach liess die ganze Seite von Anfang an).
Es waere doch moeglich, dass du Befehle an deine OpenDTU schicken kannst, von deiner Windows PC Obeflaeche.
Und wahrscheinlich die Befehle die du brauchst.
Wenn du das erreicht hast (die Befehle zu schicken und es funktioniert so wie du es haben willst), dann koenntest du den Prozess automatisieren mit einem Scheduler (siehe https://system-scheduler.informer.com/4.3/ ).
Ich habe selber nie davon gebrauch gemacht aber vielleicht hast du hiermit eine einfache Loesung ohne HA Home Assistant.
(OBWOHL ich wuerde dir waermstens empfehlen, Home Assistant zu installieren, es lohnt sich!!! siehe https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/Home-Assistant
